### ISSUE TITLE
Logs in debugger console : No handler registered for 'null-vm'

### ISSUE THREAD

**Issue by tukobadnyanoba** [docker/extensions-sdk#199]
This prevents the app from working and reactjs app has to be reloaded to make it again functioning.

**Comment by felipecruz91** [comment]
Hi @tukobadnyanoba, could you provide more information about when this happens?

Also, could you provide the output of `docker extension version` and `docker version`? Thank you.

**Comment by tukobadnyanoba** [comment]
The error occurs when the **extension window** is idle for more than 10-15 mins. Restarting extension is currently a    workaround but its hard to know when to restart the extension. I could know about this error as the **developer tools window** was open, the ui is just stuck
```
❯ docker version
Client: Docker Engine - Community
 Cloud integration: v1.0.25
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:02:57 2022
 OS/Arch:           linux/amd64
 Context:           desktop-linux
 Experimental:      true

Server: Docker Desktop 4.9.0 (80466)
 Engine:
  Version:          20.10.16
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.10
  Git commit:       f756502
  Built:            Thu May 12 09:15:42 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.4
  GitCommit:        212e8b6fa2f44b9c21b2798135fc6fb7c53efc16
 runc:
  Version:          1.1.1
  GitCommit:        v1.1.1-0-g52de29d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```
```
❯ docker extension version
Client Version: v0.2.4
Server API Version: 0.2.3
```

**Comment by tukobadnyanoba** [comment]
Full logs : Error: Error invoking remote method 'null-vm-docker-cmd': No handler registered for 'null-vm-docker-cmd'

**Comment by benja-M-1** [comment]
Hi @tukobadnyanoba to help us investigate this issue and help you as fast as possible, can you share the repo or at least the following?

* the metadata.json of your app
* the piece of code that triggers the issue in your UI
* the backend route handler
Thank you :)

**Comment by benja-M-1** [comment]
Hi @tukobadnyanoba maybe you missed my previous message? :) Let us know how you solve the issue if you don't face it again.

**Comment by benja-M-1** [comment]
Hi @tukobadnyanoba out of curiosity, do you have any redirection in JavaScript in your extension? 

I found a bug that, when there is a call to something similar to 'window.location.replace()'. In this case the extension name is missing from the query parameters. But the name is needed to register ipc handlers in the SDK which can lead to the error your are seeing.

**Comment by benja-M-1** [comment]
@tukobadnyanoba if ever you have the time to answer my previous message that would be awesome. We merged a PR that might fix it and will be available in 4.12 (by September I think) but we need your input on it.

**Comment by benja-M-1** [comment]
@tukobadnyanoba no worries, thank you for the update! I am glad we found the problem :) 

So the workaround is to ensure the two query params `extensionName` and `hasBackend`, if present, are passed in the URL when redirecting.

I close the issue then.
